apps/system/test/unit/audio_channel_controller_test.js

```diff
@@ -0,0 +1,254 @@
+/* global AudioChannelController */
+'use strict';
+
+requireApp('system/js/base_ui.js');
+requireApp('system/js/audio_channel_controller.js');
+
+suite('system/AudioChannelController', function() {
+  var subject;
+  var appAddEventStub;
+  var channelAddEventnnelStub;
+  var mockApp = {
+    instanceID: 'appID' ,
+    element: {
+      addEventListener: function() {}
+    }
+  };
+  var mockChannel = {
+    _domRequest: {},
+    name: 'channelName' ,
+    addEventListener: function() {},
+    isActive: function() {
+      return mockChannel._domRequest;
+    },
+    setMuted: function() {
+      return mockChannel._domRequest;
+    },
+    setVolume: function() {},
+    _triggerDomRequestOnsuccess: function() {
+      mockChannel._domRequest.onsuccess({
+        target: { result: true }
+      });
+    }
+  };
+
+  setup(function() {
+    appAddEventStub = this.sinon.stub(mockApp.element, 'addEventListener');
+    channelAddEventnnelStub = this.sinon.stub(mockChannel, 'addEventListener');
+    subject = new AudioChannelController(mockApp, mockChannel);
+  });
+
+  teardown(function() {
+    appAddEventStub.restore();
+    channelAddEventnnelStub.restore();
+  });
+
+  suite('Initialize the module', function() {
+    test('Register events', function() {
+      assert.ok(appAddEventStub.withArgs('_willdestroy').calledOnce);
+      assert.ok(channelAddEventnnelStub.withArgs('activestatechanged')
+        .calledOnce);
+    });
+
+    test('Audio channel is not active', function() {
+      assert.equal(subject.isActive(), false);
+    });
+
+    test('Audio channel is not playing', function() {
+      assert.equal(subject.isPlaying(), false);
+    });
+
+    test('Audio channel is not fading out', function() {
+      assert.equal(subject.isFadingOut(), false);
+    });
+
+    test('It is not vibrating', function() {
+      assert.equal(subject.isVibrating(), false);
+    });
+  });
+
+  suite('Event handlers', function() {
+    var publishStub;
+
+    setup(function() {
+      publishStub = this.sinon.stub(subject, 'publish');
+    });
+
+    teardown(function() {
+      publishStub.restore();
+    });
+
+    test('Handel activestatechanged event', function() {
+      subject.handleEvent({ type: 'activestatechanged', target: mockChannel });
+      mockChannel._triggerDomRequestOnsuccess();
+      assert.ok(publishStub.withArgs('statechanged').calledOnce);
+    });
+
+    test('Handel _willdestroy event', function() {
+      subject.handleEvent({ type: '_willdestroy' });
+      assert.ok(publishStub.withArgs('terminated').calledOnce);
+    });
+  });
+
+  suite('Deal with policies', function() {
+    var playStub;
+    var pauseStub;
+    var fadeOutStub;
+    var fadeInStub;
+    var vibrateStub;
+
+    setup(function() {
+      playStub = this.sinon.stub(subject, '_play');
```

Comment:
nit: this.sinon.spy(subject, 'play');
assert.isTrue(subject.play.called); // don't need to declare the stub* variables